### DISCUSSION TITLE
How to use the u-root version of "scp"

### DISCUSSION THREAD

**Discussion by sweetbbak** [u-root/u-root#2862] [Q&A]
Im trying to use and understand how the u-root implementation of `scp` works. Reading the code shows that it expects a response and sends a reply, then transfers the file using io.Copy()... But when I try to use this tool it just hangs indefinitely

this is my assumption of how this is supposed to be used:
```sh
./scp -f ./my_file | ssh remote_host './scp -t ./my_file2'
# but it does nothing but hang forever
```
I've done a lot of other tests but I cannot sus out how this is supposed to work. even something like "./scp -f file | ./scp -t file2" hangs forever. Im really not sure how these two things are supposed to communicate.

**Comment by hugelgupf** [comment]
I'll be honest, I'm not entirely sure as well. @rminnich got any ideas? @bluecmd as well if you're around.

**Comment by bluecmd** [comment]
I think it emulates openssh's use of scp when receiving, so you should be able to use `scp -O myfile my-u-root:target` from a machine with OpenSSH. For initiating from inside u-root I'm not sure there's a good way, you need a bidirectional pipe I'm pretty sure.

**Reply by sweetbbak** [reply]
Thank you for the insight.
